### PR TITLE
STORM-2623: Add in a whitelist for scheduler strategies

### PR DESCRIPTION
**Pull request by knusbaum**

### CODE REVIEW

File: storm-client/src/jvm/org/apache/storm/utils/ReflectionUtils.java [hunk 1]
```diff
@@ -72,6 +74,17 @@ public static ReflectionUtils setInstance(ReflectionUtils u) {
         }
     }
 
+    public static <T> T newSchedulerStrategyInstance(String klass, Map<String, Object> conf) {
+        List<String> allowedSchedulerStrategies = (List<String>) conf.get(Config.NIMBUS_SCHEDULER_STRATEGY_CLASS_WHITELIST);
+        if(allowedSchedulerStrategies == null || allowedSchedulerStrategies.contains(klass)) {
+            return newInstance(klass);
+        }
+        else {
+            //throw new RuntimeException("Cannot run topology with strategy [" + klass + "]. Allowed strategies are: " + allowedSchedulerStrategies);
```
Comment (HeartSaVioR) [inline storm-client/src/jvm/org/apache/storm/utils/ReflectionUtils.java]: Nit: Better to remove comment here.

**Comment by HeartSaVioR** [comment]
@knusbaum Sorry now there's also tiny merge conflict. Could you also address this?

**Comment by knusbaum** [comment]
@HeartSaVioR @kishorvpatil Please have a look.

**Comment by HeartSaVioR** [comment]
+1
@knusbaum If you don't mind I'd like to request squashing commits into one since it's fairly small change.

**Comment by knusbaum** [comment]
@HeartSaVioR Sounds good.

**Comment by HeartSaVioR** [comment]
@knusbaum I guess it's messed up while squashing. Could you sort it out? If commit eabfa7d is clean one please let me know that I can just cherry pick it.

**Comment by HeartSaVioR** [comment]
@knusbaum Kindly reminder.

**Pull request merged**